### DISCUSSION TITLE
"re" and "rw" in LTSP scripts

### DISCUSSION THREAD

**Discussion by maccadia** [ltsp/ltsp#198] [Q&A] [answered]
Hi,

What are "re" and "rw" in LTSP scripts ?

e.g. in ` ltsp/ltsp/client/init/45-networking.sh `
```bash
config_network_manager() {
    re test "DEVICE=$DEVICE" != "DEVICE="
    rw rm -rf /run/netplan /etc/netplan /lib/systemd/system-generators/netplan
```
I have never seen this in bash scripts and Google is of little help.

**Comment by alkisg** [comment, marked as answer]
They're functions, implemented in https://github.com/ltsp/ltsp/blob/master/ltsp/ltsp#L218

re is an abbreviation for "Run and Exit on failure", while rw for "Run and Warn on failure".

That way, debugging information is a lot better than what usually shell scripts offer: they show the command that failed, offer a shell right at the point of failure if DEBUG_SHELL is set, where a programmer can examine the content of variables etc.

**Comment by maccadia** [comment]
Nice ! Thanks
